Documentation/ceph-object-bucket-claim.md

```diff
@@ -28,7 +28,8 @@ spec:
   generateBucketName: photo-booth [4]
   storageClassName: rook-ceph-bucket [4]
   additionalConfig: [5]
-    ANY_KEY: VALUE ...
+    maxObjects: "1000"
```

Comment:
I don't understand why quota is being defined/handled in the OBC vs. in the storage class. Isn't quota something the a storage admin manages vs. individual developers? Are you letting a developer define quota values _lower_ than what is (or would be) defined in the storage class, say as a way to make sure her app doesn't "run away" inserting objects?

Reply:
@jeffvance I feel it make more sense to add in OBC than storage class, especially for green field cases. So that each user can be limited for different quotas for same bucket. If it is property of storage class then all the obc will have same quota. We can consider adding two levels of quota, one in storage class(will be bucket quota IMO) and OBC quota(user quota). But it may need thorough testing

Reply:
@thotz I did some more searching on k8s storage quotas and they are namespace based. I could not find a way to create a cluster-wide quota, so I am better understanding defining quota in the OBC.
At some point you may want to consider (in a separate pr) whether or not cluster-scoped bucket quotas make sense.

Reply:
It is still an important difference that OBC quotas can be defined by the owner of the OBC, while storage class quotas would be defined by the admin. Would OBC owners really want to impose their own quotas? It seems like more of an admin action. But if the OBC can set the quota, at least it should be ignored if it is higher than a quota defined in the storage class.

Reply:
@travisn @thotz If we have storageclass quotas then I think OBC quota should only be able to _reduce_ the storage class quota values.

Reply:
@thotz What if the storageclass quota is different from the OBC quota? I might have missed in this PR where we deal with that if there is a conflict.

Reply:
@travisn in this PR, I have not considered conflicts with respect to quota enforced on `storage class`.

Reply:
If the quota limits are higher on the OBC than on the storage class, seems like we need to prevent that, or else the user could have whatever quotas they want instead of what the admin requires.

Reply:
@jeffvance , I was trying to test this feature with storage limit specified for storage class via `resource quota`. I can see that limitation specified is not getting effect on storage class for OBC. Is this expected?

Reply:
@thotz Reading [here](https://kubernetes.io/docs/tasks/administer-cluster/limit-storage-consumption/) and [here](https://kubernetes.io/docs/concepts/policy/resource-quotas/) it looks like you can only affect PVC via resource quota. This is not an area I have direct experience with. Maybe you'll need to look at the code.

Reply:
> @thotz Reading [here](https://kubernetes.io/docs/tasks/administer-cluster/limit-storage-consumption/) and [here](https://kubernetes.io/docs/concepts/policy/resource-quotas/) it looks like you can only affect PVC via resource quota. This is not an area I have direct experience with. Maybe you'll need to look at the code.

Even my understanding was the same. Just want to confirm the same